applications/IgaApplication/custom_elements/iga_membrane_element.cpp

```diff
@@ -193,119 +193,6 @@ namespace Kratos
         KRATOS_CATCH("");
     }
 
-    void IgaMembraneElement::CalculateInitialStiffnessMatrix(
-        MatrixType& rLeftHandSideMatrix,
-        const ProcessInfo& rCurrentProcessInfo
-    )
-    {
-        KRATOS_TRY
-
-        const auto& r_geometry = GetGeometry();
-
-        // definition of problem size
-        const SizeType number_of_nodes = r_geometry.size();
-        const SizeType mat_size = number_of_nodes * 3;
-
-        const auto& r_integration_points = r_geometry.IntegrationPoints();
-
-        const IntegrationMethod integration_method = GetGeometry().GetDefaultIntegrationMethod();
-        const GeometryType::ShapeFunctionsGradientsType& r_shape_functions_gradients = GetGeometry().ShapeFunctionsLocalGradients(integration_method);
-
```

Comment:
why do you delete all this?

Reply:
In the new CalculateDampingMatrix function, we can obtain the stiffness contribution for the damping by calling the CalculateLeftHandSide directly. 

Therefore, the CalculateInitialStiffnessMatrix, which was actually a copy of the CalculateAll function but neglecting calculation of residual vector, is not needed anymore.

Reply:
gotcha. I thought that was for avoiding to store the member variables. but it seems those are pre computed.

Reply:
In the new CalculateDampingMatrix function, we can obtain the stiffness contribution for the damping by calling the CalculateLeftHandSide directly. Thus, the CalculateInitialStiffnessMatrix, which was actually a copy of the CalculateAll function but neglecting calculation of residual vector, is not needed anymore.

Reply:
but does that mean that it is not the initial stiffness matrix anymore?

Reply:
Yeah, I did a mistake. Thanks for asking this.

By using the new CalculateDampingMatrix, the stiffness matrix will be updated for each non-linear step. It is analogous to the coupling condition where we have conservative and non-conservative integration. 

I need to think more about how to compute the initial stiffness more efficiently, also for the Shell3pElement.